hyperspy/component.py

```diff
@@ -1009,15 +1009,20 @@ def __call__(self):
         -------
         numpy array
         """
-        return self.model.__call__(component_list=[self])
+        axis = self.model.axis.axis[self.model.channel_switches]
+        component_array = self.function(axis)
+        return component_array
 
     def _component2plot(self, axes_manager, out_of_range2nans=True):
         old_axes_manager = None
         if axes_manager is not self.model.axes_manager:
             old_axes_manager = self.model.axes_manager
             self.model.axes_manager = axes_manager
             self.fetch_stored_values()
-        s = self.__call__()
+        if self.model:
```

Comment:
I think that it is unnecessary here to check if model is defined because this method is private and designed to be called only from model. The rest of the code assumes that model is defined.

Reply:
Yes and I think this was already changed in my last commit https://github.com/hyperspy/hyperspy/pull/1851/commits/e884931abfc1a956c789ed534172bf22497fdc75!